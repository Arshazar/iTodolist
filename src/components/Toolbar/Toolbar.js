import React, { useState } from 'react'
import { StyledToolbar } from './Toolbar.styled'
import AddTodo from '../../containers/AddTodo'
import Undo from '../../containers/Undo'

const Toolbar = () => {
    const [modal, setModal] = useState(false)
    const toggle = () => setModal(!modal)

    return (
        <>
            <StyledToolbar>
                <button onClick={() => setModal(true)}>Add new</button>
                <Undo />
            </StyledToolbar>
            {modal ? <AddTodo modal={modal} toggle={toggle} /> : <div />}
        </>
    )
}

export default Toolbar
