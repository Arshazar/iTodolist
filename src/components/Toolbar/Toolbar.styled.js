import styled from 'styled-components'

export const StyledToolbar = styled.div`
    margin: 0;
    text-align-last: end;

    button {
        border: 0;
        border-radius: 0;
        cursor: pointer;
        border-radius: 0;
        padding: 12px;
        color: white;
        background-color: ${({ theme }) => theme.primaryButtonColour};

        &:hover {
            background-color: navy;
        }
    }
`
