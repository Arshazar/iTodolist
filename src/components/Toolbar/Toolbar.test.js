import React from 'react'
import { shallow } from 'enzyme'
import Toolbar from './Toolbar'

it('renders Toolbar component', () => {
    expect(shallow(<Toolbar />)).toMatchSnapshot()
})
