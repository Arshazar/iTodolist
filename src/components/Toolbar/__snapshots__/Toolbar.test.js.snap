// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Toolbar component 1`] = `ShallowWrapper {}`;
